sensing/pointcloud_preprocessor/src/downsample_filter/voxel_grid_downsample_filter_nodelet.cpp

```diff
@@ -64,31 +64,144 @@ VoxelGridDownsampleFilterComponent::VoxelGridDownsampleFilterComponent(
 {
   // set initial parameters
   {
-    voxel_size_x_ = static_cast<double>(declare_parameter("voxel_size_x", 0.3));
-    voxel_size_y_ = static_cast<double>(declare_parameter("voxel_size_y", 0.3));
-    voxel_size_z_ = static_cast<double>(declare_parameter("voxel_size_z", 0.1));
+    voxel_size_x_ = static_cast<float>(declare_parameter("voxel_size_x", 0.3));
+    voxel_size_y_ = static_cast<float>(declare_parameter("voxel_size_y", 0.3));
+    voxel_size_z_ = static_cast<float>(declare_parameter("voxel_size_z", 0.1));
   }
 
+  inverse_voxel_size_[0] = 1.0f / voxel_size_x_;
+  inverse_voxel_size_[1] = 1.0f / voxel_size_y_;
+  inverse_voxel_size_[2] = 1.0f / voxel_size_z_;
+  pcl::for_each_type<typename pcl::traits::fieldList<pcl::PointXYZ>::type>(
+    pcl::detail::FieldAdder<pcl::PointXYZ>(xyz_fields_));
+
```

Comment:
I'd like to hide this process in this function.
I think it would be better to have a separate class for the process of doing the voxel grid filter.
If possible, it would be nice to be able to define an Interface class to switch with the PCL voxel grid filter.